common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -553,6 +553,29 @@ public UTF8String trim() {
     return copyUTF8String(s, e);
   }
 
+  /**
+   * Trims space characters (ASCII 32) from both ends of this string.
+   *
+   * @return this string with no spaces at the start or end
+   */
+  public UTF8String nonCopyTrim() {
```

Comment:
another way is to embed the trim logic into `toInt`.

Reply:
I will check this